components/vc/front/src/cockpit_lights.c

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file cockpit_lights.c
+ * @brief Module source that manages IMD and BMS lights 
+ */
+
+/******************************************************************************
+ *                             I N C L U D E S
+ ******************************************************************************/
+
+
+#include "cockpit_lights.h"
+#include "Module.h"
+#include "ModuleDesc.h"
+#include "string.h"
+#include "drv_outputAD.h"
+#include "app_vehicleState.h"
+ 
+ 
+ /******************************************************************************
+  *                       P U B L I C  F U N C T I O N S
+  ******************************************************************************/
+
+ IMD_LIGHT_state_E IMD_LIGHT_getState(void)
+ {
+    return IMD_LIGHT_data.state;
+ }
+ 
+ CAN_digitalStatus_E IMD_LIGHT_getStateCAN(void)
+ {
+    return (IMD_LIGHT_data.state == IMD_LIGHT_ON) ?CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+ }
+ 
+ static void IMD_LIGHT_init(void)
+ {
+   IMD_LIGHT_data.state = IMD_LIGHT_OFF;
+ }
+
+ BMS_LIGHT_state_E BMS_LIGHT_getState(void)
+ {
+    return BMS_LIGHT_data.state;
+ }
+ 
+ CAN_digitalStatus_E BMS_LIGHT_getStateCAN(void)
+ {
+    return (BMS_LIGHT_data.state == BMS_LIGHT_ON) ? CAN_DIGITALSTATUS_ON : CAN_DIGITALSTATUS_OFF;
+ }
+ 
+ static void BMS_LIGHT_init(void)
+ {
+    BMS_LIGHT_data.state = BMS_LIGHT_OFF; 
+ }
+ 
+ static void BMS_LIGHT_periodic_10Hz(void)
+
+ const bool chg_valid = (CANRX_get_signal(VEH, bmsb_bmsStatusMem, &IMD_LIGHT_state_E) == CANRX_MESSAGE_VALID);
```

Comment:
`IMD_LIGHT_state_E` is not a variable, its a type. You have to define a struct like
```
struct
{
    IMD_LIGHT_state_E imd_state;
    BMS_LIGHT_state_E bms_state;
} cockpitLight_data;
```

and then
```suggestion
 const bool chg_valid = (CANRX_get_signal(VEH, bmsb_bmsStatusMem, &cockpitLight_data.bms_state) == CANRX_MESSAGE_VALID);
```

Reply:
Also, `bmsb_bmsStatusMem` needs to be `BMSB_bmsStatusMem`

Reply:
Also, you dont use the is valid here

Reply:
Also, `chg_valid` doesnt make sense here. is this anything related to the charger?